whatsagent/tests.py

```diff
@@ -0,0 +1,131 @@
+import json
+
+from django.test import TestCase
+from rest_framework.response import Response
+
+from unittest.mock import Mock, patch
+
+from rest_framework.test import APIClient
+import requests
+from typing import Dict, List
+
+
+class MockResponse:
```

Comment:
Run all tests, failed with:

```
(whatsagent) robertsgreibers@MacBook-Pro-2 whatsagent % python manage.py test -v 2 --keepdb                                                
Found 3 test(s).
Using existing test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...
Operations to perform:
  Synchronize unmigrated apps: corsheaders, django_extensions, messages, rest_framework, staticfiles
  Apply all migrations: admin, api, auth, contenttypes, sessions
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying api.0001_initial... OK
  Applying api.0002_rename_is_active_user_terms_user_name... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying sessions.0001_initial... OK
System check identified no issues (0 silenced).
test_failed_to_send_email (whatsagent.tests.TestSignup) ... First argumant bashobi
Second argumant True
FAIL
test_successful_signup (whatsagent.tests.TestSignup) ... [<Mock id='4570454672'>]
First argumant bashobi
Second argumant True
ok
test_unaccepted_terms_checkbox (whatsagent.tests.TestSignup) ... ok

======================================================================
FAIL: test_failed_to_send_email (whatsagent.tests.TestSignup)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/robertsgreibers/projects/1_mentoring_projects/2_bashobi/whatsagent/whatsagent/tests.py", line 91, in test_failed_to_send_email
    self.assertEqual(res.status_code, 503)
AssertionError: 201 != 503

----------------------------------------------------------------------
Ran 3 tests in 2.475s

FAILED (failures=1)
Preserving test database for alias 'default' ('file:memorydb_default?mode=memory&cache=shared')...
(whatsagent) robertsgreibers@MacBook-Pro-2 whatsagent % 
```

Reply:
@basroy still fails, need to make mocked test adjustments

Reply:
@basroy signup is missing some test cases

Expected:
![signup-tests-expected](https://user-images.githubusercontent.com/12160621/203095558-c3b0cbae-663f-4ded-ae21-83da1c5c6cb6.png)

Yours:
![Screenshot 2022-11-21 at 17 37 21](https://user-images.githubusercontent.com/12160621/203095621-e947d3f0-41bf-4cee-bcc5-73948350a8c2.png)